GitHubContents.ps1

```diff
@@ -508,11 +556,37 @@ filter Add-GitHubContentAdditionalProperties
 
         if (-not (Get-GitHubConfiguration -Name DisablePipelineSupport))
         {
-            $elements = Split-GitHubUri -Uri $item.url
+            if ($item.url)
+            {
+                $uri = $item.html_url
+            }
+            else
+            {
+                $uri = $item.content.html_url
+            }
+
+            $elements = Split-GitHubUri -Uri $uri
             $repositoryUrl = Join-GitHubUri @elements
+
             Add-Member -InputObject $item -Name 'RepositoryUrl' -Value $repositoryUrl -MemberType NoteProperty -Force
+
+            $hostName = $(Get-GitHubConfiguration -Name 'ApiHostName')
+
+            if ($uri -match "^https?://$hostName/?([^/]+)/?([^/]+)/blob/?([^/]+)/?([^#]*)?$")
+            {
+                $branchName = $Matches[3]
+                $path = $Matches[4]
```

Comment:
With the regex change above, you'll need to update these match references as well:

```suggestion
                $branchName = $Matches[4]
                $path = $Matches[5]
```